cpp/src/structure/graph.cu

```diff
@@ -32,7 +34,9 @@ void degree_from_offsets(vertex_t number_of_vertices,
 }
 
 template <typename vertex_t, typename edge_t>
-void degree_from_vertex_ids(edge_t number_of_edges,
+void degree_from_vertex_ids(const cugraph::experimental::Comm& comm,
+                            vertex_t number_of_vertices,
+                            edge_t number_of_edges,
                             vertex_t const *indices,
                             edge_t *degree,
```

Comment:
And with OPG, shouldn't we better distribute degree? If every process has the degree values for the entire set of vertices, we cannot run this unless the entire set of vertices fits in a single GPU.

Reply:
We should, as part of 2D partitioning. Just focussing on the shortest path for plugging in the 1D PageRank for now and identifying obstacles across the stack/product. There should be more incremental OPG work ( through RAFT and 2D) before all this reaches the "final" version.